core/src/main/java/ai/timefold/solver/core/api/score/stream/common/ConcurrentUsage.java

```diff
@@ -1,11 +1,15 @@
-package ai.timefold.solver.examples.common.experimental.api;
+package ai.timefold.solver.core.api.score.stream.common;
 
-public interface IntervalCluster<Interval_, Point_ extends Comparable<Point_>, Difference_ extends Comparable<Difference_>>
+public interface ConcurrentUsage<Interval_, Point_ extends Comparable<Point_>, Difference_ extends Comparable<Difference_>>
         extends Iterable<Interval_> {
     int size();
 
     boolean hasOverlap();
 
+    int getMinimumConcurrentUsage();
+
+    int getMaximumConcurrentUsage();
```

Comment:
The first method speak of overlap.
The other two methods speak of concurrent usage.
We should standardize this; personally, `minimumOverlap` is more self-explanatory to me.